scripts/metadata_comparison/metadata_comparison/comparer.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+#
+# comparer.py
+#
+# Purpose: Compare performance metadata JSON files produced by Digester and produce result in CSV format
+#
+# Usage: python3 comparer.py [-h] [-v] [--json_paths JSONPATH [JSONPATH ...]] [--output_path OUTPUTPATH]
+#
+# Python Prereqs (at least, the ones which I needed to manually install... YMMV):
+#
+#   * pip3 install --upgrade pandas
+#
+
+import argparse
+import json
+import pandas
+from pathlib import Path
+from metadata_comparison.lib.logging import *
+
+logger = log.getLogger('metadata_comparison.comparer')
+
+def read_json_files(*paths):
+    """
+    Produces list of tuples: [[path, json], [path, json], ...]
+    """
+    jsons = []
+    for path in paths:
+        with open(path,) as file:
+            jsons.append([path, json.load(file)])
+
+    return jsons
+
+
+def compare_jsons(*pathsAndJsons):
+    """
+    Uses pandas library to convert JSONs into into dataframes, and concatenate those dataframes into a single one.
+    Performs sanity check, producing exception, if at least one of the JSONs doesn't have matching subset of keys.
+    """
+    columnToCompareNameEnding = ".overallRuntimeSeconds"
+    result = pandas.DataFrame()
+    lastCols = []
+    for pathAndJson in pathsAndJsons:
+        df = pandas.json_normalize(pathAndJson[1])
+        cols = [c for c in df.columns if c[len(c)-len(columnToCompareNameEnding):] == columnToCompareNameEnding or c == 'version']
+
+        if lastCols and lastCols != cols:
```

Comment:
Is this resilient to re-orderings in the json (eg if the calls get reordered in the json, does this comparison still match)? Maybe worth a test case even if pandas makes sure it always works?

Reply:
Good catch. Thank you.